site-src/concepts/guidelines.md

```diff
@@ -111,4 +111,13 @@ cases. One variation from the Kubernetes conventions is for fields with
 default values. These fields are considered optional but are not pointers
 since they will never be empty.
 
+Another convention this project uses is for plural field names for lists
+in our CRDs. We use the following rules:
+
+- If the field name is a noun, use a plural value.
+- If the field name is a verb, use a singular value.
+
+So for example, in HTTPRoute, `hostnames` uses a plural, but `ForwardTo` is singular,
```

Comment:
Nit: capitalization is inconsistent here. I think I'd lean towards switching to `forwardTo` instead of switching to `Hostnames` but don't have a strong preference.